### PR TITLE
nss-mdns: 0.10.0 -> 0.15.1

### PR DESCRIPTION
**Pull request by graysonhead**
###### Description of changes

To be forward, this PR may not be suitable to merge in its current state, it probably needs review from someone who was involved in the MUSL patch made to 0.10.0 (@r-burns or someone else) to ensure this newer version doesn't break anything for non-glibc systems.

Additionally I'd like to call attention to the fact that, this version is a fork of the original, so the source repo changed. 

For context, this PR fixes the issue I experienced in #180883. 

###### Things done

- Built on platform(s)
  - [ x ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-burns** [comment]
I think it's fine to drop the musl patch here. Openembedded dropped their patch anyway, stating:

> There's no point in building an NSS module for a C library that doesn't support NSS.

https://github.com/openembedded/openembedded-core/commit/e3197eeaf352cb5c578a675ce2a07881fad597e6

File: pkgs/tools/networking/nss-mdns/default.nix
Comment (r-burns) [inline pkgs/tools/networking/nss-mdns/default.nix:1]:
```suggestion
{ fetchurl, lib, autoreconfHook, pkg-config, stdenv, fetchpatch, fetchFromGitHub }:
```

Do not pass all of pkgs here, pass individual packages as required

File: pkgs/tools/networking/nss-mdns/default.nix
Comment (r-burns) [inline pkgs/tools/networking/nss-mdns/default.nix:15]: Remove trailing whitespace to fix editorconfig check

**Review by r-burns — CHANGES_REQUESTED** [review]

**Comment by graysonhead** [comment]
@r-burns Thanks for the feedback, I've made the changes you suggested.

File: pkgs/tools/networking/nss-mdns/default.nix
Comment (smancill) [inline pkgs/tools/networking/nss-mdns/default.nix:11]:
```suggestion
    hash = "sha256-iRaf9/gu9VkGi1VbGpxvC5q+0M8ivezCz/oAKEg5V1M=";
```

Nixpkgs is slowly moving to SRI hashes, thus the `hash` attribute now.

**Comment by r-burns** [comment]
Changes look good. Using a fork of the original package warrants some scrutiny, but repology shows that this fork is commonly used on major distros, so I think we are fine here.

It appears there are no maintainers for this package for me to ping, so I will leave a couple days' grace period before merging.

**Comment by smancill** [comment]
Result of `nixpkgs-review pr 182236` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>keylight-controller-mschneider82</li>
  </ul>
</details>

**Comment by smancill** [comment]
Please use rebase to squash the original first commit, as explained by the docs:

https://nixos.org/manual/nixpkgs/stable/#submitting-changes-hotfixing-pull-requests

**Review by r-burns — APPROVED** [review]

**Comment by ilya-fedin** [comment]
> so I will leave a couple days' grace period before merging

How long? :thinking:  I have the problem with nss-mdns, too...

**Comment by r-burns** [comment]
Thanks for the ping, wish I could tell github to auto-remind me when I've approved something.

**Pull request merged**
